lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/IO/DB.hs

```diff
@@ -54,75 +51,26 @@ withSqlContextInMemory
     -> (SqlContext -> IO a)
     -- ^ Action to run
     -> IO a
-withSqlContextInMemory tr action = do
-    withDBHandleInMemory tr $ \dbhandle -> do
-        -- Lock ensures that database operations are sequenced.
-        lock <- newMVar (dbBackend dbhandle)
-        let runSqlM :: SqlM a -> IO a
-            runSqlM cmd =
-                withMVarMasked lock (observe . Persistent.runSqlConn cmd)
-        action $ SqlContext{runSqlM}
-  where
-    observe :: IO a -> IO a
-    observe = bracketTracer (contramap MsgRun tr)
+withSqlContextInMemory tr = withSqliteFile tr ":memory:"
```

Comment:
so in the first version we want to be in-memory, correct?

Reply:
And later add, part file-persistence + part in-memory regime?

Reply:
Yes, the first version of the Deposit Wallet will be in-memory. However, for this, we won't use the `SqlContext` created by `withSqlContextInMemory` — since everything is in memory, we can use regular Haskell data types and don't have to deal with SQL at all.

That is why I would like to postpone testing these functions for now — in the immediate future, we will treat `withSqlContextInMemory` as a fancy `id` function.